app/models/setting.rb

```diff
@@ -198,28 +198,29 @@ def parse_string_value(val)
     true
   end
 
-  def self.create(opts)
+  # in order to avoid code duplication, this method was introduced
+  def self.create_find_by_name(opts, &block)
```

Comment:
You may remove the `&block`, implicit is way faster - you may get some background here https://github.com/ruby/ruby/pull/1240

Reply:
keep on forgetting that :( thanks